deepchem/molnet/load_function/chembl_datasets.py

```diff
@@ -103,18 +106,21 @@ def load_chembl(shard_size=2000,
     for transformer in transformers:
       dataset = transformer.transform(dataset)
 
-  splitters = {
-      'index': deepchem.splits.IndexSplitter(),
-      'random': deepchem.splits.RandomSplitter(),
-      'scaffold': deepchem.splits.ScaffoldSplitter()
-  }
+  if spit == None:
+    return chembl_tasks, (dataset, None, None), transformers
+  else:
+    splitters = {
+        'index': deepchem.splits.IndexSplitter(),
+        'random': deepchem.splits.RandomSplitter(),
+        'scaffold': deepchem.splits.ScaffoldSplitter()
+    }
 
-  if split in splitters:
-    splitter = splitters[split]
-    print("Performing new split.")
-    train, valid, test = splitter.train_valid_test_split(dataset)
+    if split in splitters:
```

Comment:
why are we checking if split is in splitters here?  We have already done the None check.  If it is not in splitters we will raise a really weird exception about not initializing train, valid, test.  Where as if you don't do the check the exception raised will make far more sense to the programmer.

Reply:
I understand you didn't put in this code but we should leave it better than we found it now that we are here :smile: